### DISCUSSION TITLE
[Android] Bundle build fails. No jni directory generated for node_modules

### DISCUSSION THREAD

**Discussion by einjunge99** [reactwg/react-native-new-architecture#254] [Q&A]
As the title suggests, I'm struggling to generate a bundle build for my project. While my project has a lot of dependencies, I managed to create a minimal replicate repository, with Bugsnag as the only dependency installed (You can check the issue here: [https://github.com/bugsnag/bugsnag-js/issues/2273](https://github.com/bugsnag/bugsnag-js/issues/2273)). 

The thing is that it seems weird to me that no one has ever reported something similar and even at some point I managed to get a successful build creating a project following the exact same steps as the minimal replicate repository, pointing that it may be something related to my local setup, but even If that were the case, I got the exact same issue while running my CI pipeline.

Is this a known issue? Any help would be gladly accepted.

**Comment by cortinico** [comment]
1. What is the error message?
2. What are the exact steps to reproduce your problem?

**Reply by einjunge99** [reply]
Thanks for you response @cortinico!

**1. What is the error message?**

> Execution failed for task ':app:configureCMakeRelWithDebInfo[arm64-v8a]'.

Upon further inspection, as the logs attached below states, no `/Users/isaacmaldonado/Documents/Projects/Template/node_modules/@bugsnag/react-native/android/build/generated/source/codegen/jni/` were generated.


<details><summary>Error messages:</summary>

```
> Task :app:configureCMakeDebug[arm64-v8a] FAILED
C/C++: CMake Error at /Users/isaacmaldonado/Documents/Projects/Template/android/app/build/generated/autolinking/src/main/jni/Android-autolinking.cmake:9 (add_subdirectory):
C/C++:   add_subdirectory given source
C/C++:   "/Users/isaacmaldonado/Documents/Projects/Template/node_modules/@bugsnag/react-native/android/build/generated/source/codegen/jni/"
C/C++:   which is not an existing directory.
C/C++: Call Stack (most recent call first):
C/C++:   /Users/isaacmaldonado/Documents/Projects/Template/node_modules/react-native/ReactAndroid/cmake-utils/ReactNative-application.cmake:86 (include)
C/C++:   CMakeLists.txt:31 (include)
C/C++: CMake Error at /Users/isaacmaldonado/Documents/Projects/Template/node_modules/react-native/ReactAndroid/cmake-utils/ReactNative-application.cmake:89 (target_link_libraries):
C/C++:   Cannot specify link libraries for target
C/C++:   "react_codegen_BugsnagReactNativeSpec" which is not built by this project.
C/C++: Call Stack (most recent call first):
C/C++:   CMakeLists.txt:31 (include)
```
</details>

**2. What are the exact steps to reproduce your problem?**
1. Create new react native project with `npx @react-native-community/cli@latest init <APP_NAME> --version 0.76.2`
2. Follow [react native manual setup guide](https://docs.bugsnag.com/platforms/react-native/react-native/manual-setup/)
3. `cd android` folder and run `./gradlew bundleRelease`
4. See error

### Environment
* MacOS (M1)
* Bugsnag version: 8.1.3
* NodeJS: 20.15.1
* Java: zulu-17.46.19

### Example Repo
- https://github.com/einjunge99/template (Remember to replace YOUR_BUGSNAG_API_KEY with your own Bugsnag API key on `android/app/src/main/AndroidManifest.xml`)

**Reply by mikemilla** [reply]
Getting a similar issue with a package we build that contains Native Views

1. New project running 0.76+
2. `yarn add` our package: https://github.com/trycourier/courier-react-native
3. See the error when running Android

```
> Task :app:configureCMakeDebug[arm64-v8a] FAILED
C/C++: CMake Error at /path/to/sample-apps/android/app/build/generated/autolinking/src/main/jni/Android-autolinking.cmake:9 (add_subdirectory):
C/C++:   add_subdirectory given source
C/C++:   "/path/to/sample-apps/node_modules/@trycourier/courier-react-native/android/build/generated/source/codegen/jni/"
C/C++:   which is not an existing directory.
C/C++: Call Stack (most recent call first):
C/C++:   /path/to/sample-apps/node_modules/react-native/ReactAndroid/cmake-utils/ReactNative-application.cmake:85 (include)
C/C++:   CMakeLists.txt:31 (include)
C/C++: CMake Error at /path/to/sample-apps/node_modules/react-native/ReactAndroid/cmake-utils/ReactNative-application.cmake:88 (target_link_libraries):
C/C++:   Cannot specify link libraries for target
C/C++:   "react_codegen_CourierReactNativeView" which is not built by this project.
C/C++: Call Stack (most recent call first):
C/C++:   CMakeLists.txt:31 (include)
```

**Reply by cortinico** [reply]
@einjunge99 that's most likely a bug with Bugsnag. I saw the issue issue you opened for them: 
- https://github.com/bugsnag/bugsnag-js/issues/2273

Surprisingly, I was able to build your reproducer successfully so I'm wondering if that has anything to do with your local environment.

@mikemilla similarly, that seems like a bug for `@trycourier/courier-react-native`. Have you tried reporting the issue to them?

**Reply by einjunge99** [reply]
Thanks for the update @cortinico! Do you happen to know any tips for troubleshooting? Currently I have two projects, the one I uploaded at https://github.com/einjunge99/template and another one that is exactly the same, but It's successfully building at the moment.

**Reply by cortinico** [reply]
> Currently I have two projects, the one I uploaded at [einjunge99/template](https://github.com/einjunge99/template) and another one that is exactly the same, but It's successfully building at the moment.

I mentioned that `einjunge99/template` does build correctly for me. It seems like that is not the case for you right?

**Reply by einjunge99** [reply]
> It seems like that is not the case for you right?

Thats correct @cortinico. Just to clarify, by build I mean running `./gradlew bundleRelease` command. The app runs just fine on my physical device/simulator.

**Reply by williamsrosterfy** [reply]
> > It seems like that is not the case for you right?
> 
> Thats correct @cortinico. Just to clarify, by build I mean running `./gradlew bundleRelease` command. The app runs just fine on my physical device/simulator.

same issue

**Reply by einjunge99** [reply]
Is there anything that I could provide to you to enlighten this error @cortinico?

**Reply by cortinico** [reply]
@einjunge99 I was able to investigate the issue a bit further, I will answer on the linked issue you opened on Bugsnag 👍
